tools/quake3/q3map2/bsp.cpp

```diff
@@ -698,11 +698,13 @@ int BSPMain( int argc, char **argv ){
 	strClear( tempSource );
 
 	/* set standard game flags */
+	maxLMSurfaceVerts = game->maxLMSurfaceVerts;
 	maxSurfaceVerts = game->maxSurfaceVerts;
 	maxSurfaceIndexes = game->maxSurfaceIndexes;
 	emitFlares = game->emitFlares;
 	texturesRGB = game->texturesRGB;
 	colorsRGB = game->colorsRGB;
+	keepLights = game->keepLights;
```

Comment:
This affects quite much games. What are effects? Don't they handle this as RT lights?
Looks like not evey per game setting was really supposed to be used as a setting. Possibly default behavior is wished for end users.
Semantically it's clean to either use an option, or to remove it from game options of course, but this involves knowing what are effects and what users expect.

Reply:
If a game set keeplights to true in its descriptor it's expected to keep the light entities by default. I mean, that's what the game definitios are for, isn't it?

That's what we expected when we set them like that, looking for the future. Now that we wanted to use them with the RTL option we found out none of the maps kept the light entities all this time. It only works if you explicitly add the keeplights switch in the command line or worldspawn.

Reply:
As i'v said, semantically game description is for this, but setting might have been added by some screwed coder, not even caring about using freshly added setting.
Thus the question is practical effects of this. Does this only affect the case, when every of these games tries to use optional RTL?

Reply:
The only effect on the keeplights one is that the light entities are kept in the bsp file instead of deleted as default. AFAIK this is transparent for all games since undefined entities are simply ignored at spawn. And in any case, as long as that game has that value defined as false it's definitely unafected.

Reply:
Yeah, this looks like nuff safe improvement, unless it alters some expected behavior (e.g. using RTL along with lightmaps, if lights are present (doubtfully this could be actual desing tho)).

Reply:
For reference, this 76dd045925622ea3b35204df18f7fbd179b2ec51 introduced ignoring `game->keepLights`

Reply:
I've been using the console switch, yeah. I just want to stop needing to usse it :)

We are doing it for using RTL with lightmaps, btw. We are using q3map2 to compile the radiosity and the light entities kept for RTL. I just started modding a q3map2 copy to have the option to bake static lights into the lightmap if a specific flag is set, etc.

Reply:
Some downside is you can only disable by _keeplights, not -keeplights, not deadly tho.

Reply:
In times of iD it might be a speedup to avoid parsing a few hundreds of extra entities i guess.
Btw: hopefully max_entities game check occurs after these are pruned...